### PR TITLE
[libxmlmm ] Add new port libxmlmm 0.6.0

### PR DESCRIPTION
**Pull request by rioki**
**Describe the pull request**

- What does your PR fix? Fixes #

Adds libxmlmm in version 0.6.0

- Which triplets are supported/not supported? Have you updated the CI baseline?

Known good triplets:
    x86-windows
    x64-windows
    x64-linux

The code is quite portable and most should be suported.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

I checked and hope I did all right.

### CODE REVIEW

File: ports/libxmlmm/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
+include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/libxmlmm/portfile.cmake:1]:
```suggestion
```
This function is not needed here. Please remove this line.

File: ports/libxmlmm/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,59 @@
+cmake_minimum_required(VERSION 3.8)
+project(libxmlmm CXX)
+
+set(CMAKE_SHARED_LIBRARY_PREFIX)
+set(CMAKE_STATIC_LIBRARY_PREFIX)
+
+find_package(libxml2 REQUIRED)
```
Comment (NancyLi1013) [inline ports/libxmlmm/CMakeLists.txt:7]:
```suggestion
find_package(LibXml2 REQUIRED)
```
It seems that we should use this name.

The usage for `libxml2` is like this:
```
find_package(LibXml2 REQUIRED)
target_include_directories(main PRIVATE ${LIBXML2_INCLUDE_DIR})
target_link_libraries(main PRIVATE ${LIBXML2_LIBRARIES})
```

File: ports/libxmlmm/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	libxmlmm/Element.cpp
+	libxmlmm/LibXmlSentry.cpp
+	libxmlmm/Node.cpp
+	libxmlmm/ProcessingInstruction.cpp
+	libxmlmm/Text.cpp
+	libxmlmm/utils.cpp
+)
+
+add_library(libxmlmm ${SOURCES_LIBXMLMM})
+include_directories(${LIBXML2_INCLUDE_DIRS})
```
Comment (NancyLi1013) [inline ports/libxmlmm/CMakeLists.txt:41]:
```suggestion
include_directories(${LIBXML2_INCLUDE_DIR})
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/libxmlmm/CMakeLists.txt (hunk 2 above)
Comment (rioki) [reply ports/libxmlmm/CMakeLists.txt]: Yes, I was wondering why the linux build failed. Sorry I figured it out before looking at this PR.

File: ports/libxmlmm/CMakeLists.txt (hunk 3 above)
Comment (rioki) [reply ports/libxmlmm/CMakeLists.txt]: I am never sure which is the right one, but it appears both work. (Or the header is in path and it just works by chance.)

**Comment by rioki** [comment]
Do you want to squash the change into one clean comit?

File: ports/libxmlmm/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (NancyLi1013) [inline ports/libxmlmm/portfile.cmake:1]:
```suggestion
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @rioki 
Thanks for this PR.
There is no need to squash the change into one clean commit. 
The current status looks good.

Is there anything need to be done on your sides?

**Review by NancyLi1013 — APPROVED** [review]

**Comment by rioki** [comment]
Thank you for your assistance.

**Comment by NancyLi1013** [comment]
LGTM now. Thanks for your hard work.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**